### PR TITLE
Add NS custom reports

### PR DESCRIPTION
**Pull request by Achoobert**
I am not sure that the reports are 100% accurate and working, however, I reached a point of simply not having the ability to make such complex test data. 
Once NS_apps make it to staging we will have a feedback loop for finalizing tests for these.

This has to be merged in in order for automated tests to be run

### CODE REVIEW

File: config1/local.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
+/**
+ * Local environment settings
+ *
```
Comment (nh758) [inline config1/local.js:3]: This file shouldn't be committed (see https://sailsjs.com/documentation/concepts/configuration/the-local-js-file). Also this file here wouldn't be used, as the runtime > `config/local.js` created by the installer is pulled into the config container shared by all our services

File: handlers/report.js [hunk 2]
```diff
@@ -43,10 +52,20 @@ module.exports = {
          // get the AB for the current tenant
          const AB = await ABBootstrap.init(req);
          const key = req.param("reportKey");
+         const languageCode =
+            req._user.languageCode || req.param("languageCode") || "en";
+
+         // is this needed?
+         if (languageCode == "zh-hans") {
+            languageCode = "zh";
```
Comment (nh758) [inline handlers/report.js:60]: `langauageCode` is a `const`. Can change to `let` if we need to reassign.

File: reports/abReportBalance.js [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+      );
+
+      /**
+       * {
+       *    rcName1: sum of balances,
+       *    rcName2: sum of balances,
+       *    ...
+       * }
+       */
+      let rcHash = {};
```
Comment (nh758) [inline reports/abReportBalance.js:436]: In v2 Johnny wants variables to commented after the declaration not before.
```js
      let rcHash = {};
       /**
       * @const {object} rcHash
       * {
       *    rcName1: sum of balances,
       *    rcName2: sum of balances,
       *    ...
       * }
       */

**Review by nh758 — CHANGES_REQUESTED** [review]

File: reports/abReportBalance.js (hunk 3 above)
Comment (Achoobert) [reply reports/abReportBalance.js:436]: Does that pattern also apply to functions?

**Pull request merged**